### PR TITLE
fix(queryset): make distinct().count() ignore join-duplicated rows

### LINKED ISSUE #2178: count() 方法在多对多条件查询distinct()后，会返回所有的数，不是distinct()后实际的数量
**Describe the bug**
多条件复合查询，含多对多查询，调用distinct()后再使用count()统计数

**To Reproduce**
应该为distinct()后打印出来的实列数

**Expected behavior**
实际返回了所有多对多的匹配结果，没有去处重复

**Additional context**
python3.12
tortoise-orm[asyncpg]==0.25.3

### PR DESCRIPTION
**Pull request by Sanjays2402**
Closes #2178

## Description
`QuerySet.count()` always emitted `COUNT(*)`, which counts every row the query produces. `distinct()` only sets `SELECT DISTINCT` on the row query, so when a filter spans an m2m (or any other row-multiplying join) the returned objects were de-duplicated but the count was not. `CountQuery` now receives the `_distinct` flag and counts distinct primary keys; grouped queries keep the existing windowed `COUNT(*)`.

## Motivation and Context
Fixes #2178: `Model.filter(m2m_field__name__in=[...]).distinct()` returns 1 object but `.count()` returned 3.

## How Has This Been Tested?
Added `test_distinct_count_with_m2m_join` in `tests/test_queryset.py`. It fails on `develop` (`assert 3 == 1`) and passes with the fix. Full suite run on SQLite is unchanged otherwise (1789 -> 1790 passed, same 3 pre-existing environment failures before and after).

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added the changelog accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -26,6 +26,7 @@ Fixed
 - ``QuerySet.count()`` now matches the limited query result for the LIMIT/OFFSET edge cases: it returns ``0`` (instead of a negative number) when ``offset()`` exceeds the total row count, and ``0`` (instead of the total) for ``limit(0)``. (#2208)
 - Field declarations on models now resolve to their concrete type (e.g. ``CharField[str]``) in Pyright/Pylance instead of ``Field[Unknown]``; the ``Field.__new__`` type-check stub now returns ``Self``. (#2216)
 - ``TransactionContext`` now returns a ``TransactionalDBClient`` instead of a raw database connection. This change gives the correct inferred type for the transaction context. (#2232)
+- ``QuerySet.distinct().count()`` no longer counts rows duplicated by a join (e.g. filtering on a m2m relation); it now counts distinct primary keys and matches the number of rows the query returns. (#2178)
```
Comment (waketzheng) [inline CHANGELOG.rst:29]: Please move this under version 1.1.9 and update the PR reference from `#2178` to `#2255`.

**Comment by Sanjays2402** [comment]
Thanks @waketzheng — done. The entry is now under a new 1.1.9 section (Fixed) and references #2255 instead of #2178.

**Review by waketzheng — APPROVED** [review]
